packages/edit-site/src/components/global-styles/font-library-modal/context.js

```diff
@@ -192,35 +203,96 @@ function FontLibraryProvider( { children } ) {
 		return getActivatedFontsOutline( source )[ slug ] || [];
 	};
 
-	async function installFont( font ) {
+	async function installFont( fontFamilyToInstall ) {
 		setIsInstalling( true );
 		try {
-			// Prepare formData to install.
-			const formData = makeFormDataFromFontFamily( font );
+			// Get the font family if it already exists.
+			let installedFontFamily = await fetchGetFontFamilyBySlug(
+				fontFamilyToInstall.slug
+			);
+
+			// Otherwise create it.
+			if ( ! installedFontFamily ) {
+				// Prepare font family form data to install.
+				installedFontFamily = await fetchInstallFontFamily(
+					makeFontFamilyFormData( fontFamilyToInstall )
+				);
+			}
+
+			// Collect font faces that have already been installed (to be activated later)
+			const alreadyInstalledFontFaces =
+				installedFontFamily.fontFace.filter( ( fontFaceToInstall ) =>
+					checkFontFaceInstalled(
+						fontFaceToInstall,
+						fontFamilyToInstall.fontFace
+					)
+				);
+
+			// Filter out Font Faces that have already been installed (so that they are not re-installed)
+			fontFamilyToInstall.fontFace = fontFamilyToInstall.fontFace.filter(
+				( fontFaceToInstall ) =>
+					! checkFontFaceInstalled(
+						fontFaceToInstall,
+						installedFontFamily.fontFace
+					)
+			);
+
 			// Install the fonts (upload the font files to the server and create the post in the database).
-			const response = await fetchInstallFont( formData );
-			const fontsInstalled = response?.successes || [];
-			// Get intersecting font faces between the fonts we tried to installed and the fonts that were installed
-			// (to avoid activating a non installed font).
-			const fontToBeActivated = getIntersectingFontFaces(
-				fontsInstalled,
-				[ font ]
+			let sucessfullyInstalledFontFaces = [];
+			let unsucessfullyInstalledFontFaces = [];
+			if ( fontFamilyToInstall.fontFace.length > 0 ) {
+				const response = await batchInstallFontFaces(
+					installedFontFamily.id,
+					makeFontFacesFormData( fontFamilyToInstall )
+				);
+				sucessfullyInstalledFontFaces = response?.successes;
+				unsucessfullyInstalledFontFaces = response?.errors;
+			}
+
+			const detailedErrorMessage = unsucessfullyInstalledFontFaces.reduce(
+				( errorMessageCollection, error ) => {
+					return `${ errorMessageCollection } ${ error.message }`;
+				},
+				''
 			);
-			// Activate the font families (add the font families to the global styles).
-			activateCustomFontFamilies( fontToBeActivated );
+
+			// If there were no successes and nothing already installed then we don't need to activate anything and can bounce now.
+			if (
+				sucessfullyInstalledFontFaces.length === 0 &&
+				alreadyInstalledFontFaces.length === 0
+			) {
+				throw new Error(
+					__( 'No font faces were installed. ' ) +
+						detailedErrorMessage
```

Comment:
This should use sprintf and use translation placeholders.

Reply:
Thanks! Addressed this in a separate PR here: https://github.com/WordPress/gutenberg/pull/57961/commits/37f31520f905a6450e756c2f8003474f92a51880